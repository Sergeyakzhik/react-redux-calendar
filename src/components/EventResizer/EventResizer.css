.table-month .resizer {
  position: absolute;
  right: -17px;
  width: 59px;
  height: 100%;
  top: 0;
  cursor: e-resize;
  z-index: 200;
  background: #f9000080;
}

.table-month .resizer:active {
  position: absolute;
  width: 480px;
  height: 350px;
  top: -145px;
}

.table-week .resizer, .table-day .resizer {
  position: absolute;
  width: 100%;
  height: 10px;
  top: 0;
  cursor: s-resize;
  z-index: 200;
  background: #f9000080;
}

.table-week .resizer:active, .table-day .resizer:active {
  position: absolute;
  width: 480px;
  height: 350px;
  top: -145px;
}

.table-day .resizer:active {
  width: 780px;
  left: -70px;
}
