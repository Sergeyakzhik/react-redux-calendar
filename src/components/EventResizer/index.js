import React from 'react';
import './style.css';

const EventResizer = props => (
  <div
    className="resizer"
    targetKey={props.targetKey}
    onMouseDown={props.onMouseDown}
    onMouseUp={props.onMouseUp}
    onMouseLeave={props.onMouseLeave}
    onClick={e => e.stopPropagation()}
    style={props.style}
  />
);

export default EventResizer;
