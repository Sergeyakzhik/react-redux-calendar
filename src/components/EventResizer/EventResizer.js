import React from 'react';
import './EventResizer.css';

let EventResizer = props => (
  <div
    className="resizer"
    targetKey={props.targetKey}
    onMouseDown={props.onMouseDown}
    onMouseUp={props.onMouseUp}
    onClick={e => e.stopPropagation()}
    style={props.style}
  >
  </div>
);

export default EventResizer;
