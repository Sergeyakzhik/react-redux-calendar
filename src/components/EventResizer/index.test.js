import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

import EventResizer from './index';

Enzyme.configure({ adapter: new Adapter() })

describe('EventResizer', () => {
  let props = {
    style: {},
    targetKey: '',
    onMouseUp: () => {},
    onMouseDown: () => {},
    onMouseLeave: () => {},
    onClick: () => {},
  }

  let component = shallow(<EventResizer {...props} />);

  it('renders EventResizer component', () => {
    expect(component).toMatchSnapshot();
  })
})
