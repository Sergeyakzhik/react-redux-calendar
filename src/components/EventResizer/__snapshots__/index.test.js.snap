// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventResizer render EventResizer component 1`] = `
<div
  className="resizer"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={Object {}}
  targetKey=""
/>
`;
